app/controllers/venues_controller.rb

```diff
@@ -1,7 +1,8 @@
 class VenuesController < ApplicationController
-
   def show
     @venue = Venue.find_by(id: params[:id])
-    @events = @venue.active_events
+    possible_event_ids = Event.where(venue_id: @venue.id).uniq.map(&:id)
+    @items = Item.available.active.where.not(id: session[:cart]).where(event_id: possible_event_ids)
+    @events = @items.map(&:event).uniq
```

Comment:
I FINALLY got this to work but I'll need help take some of the logic out of the controller